mislabeled/probe/_linear.py

```diff
@@ -135,10 +267,12 @@ def linearize_trees(
     leaves = OneHotEncoder().fit_transform(estimator.apply(X).reshape(X.shape[0], -1))
     if is_classifier(estimator):
         linear = default_linear_model["classification"]
+        linear.fit(leaves, y)
+        return linearize_linear_model_logreg(linear, leaves, y)
     else:
         linear = default_linear_model["regression"]
-    linear.fit(leaves, y)
-    return linearize_linear_model(linear, leaves, y)
+        linear.fit(leaves, y)
+        return linearize_linear_model_ridge(linear, leaves, y)
```

Comment:
you could call directly linearize(linear, leaves, y) and the dispatch will call the right method automatically.